File: level1/javascript/콜라츠 추측 12943.js
function solution(num) {
    let cnt = 0;
    
    while(num > 1 && cnt < 500){
        num = (num%2 == 0) ? num/2 : num*3+1;
        cnt++;
    }
    return (num == 1) ? cnt : -1; // 단, 500번 반복해도 1이 되지 않은 경우 –1 반환
}

// 테스트 코드
console.log(solution(6)); // 8
console.log(solution(16)); // 4
console.log(solution(626331)); // -1

File: level1/kotlin/콜라츠 추측 12943.kt
class Solution {
    fun solution(num: Int): Int {
        var value = num.toLong()
        var count = 0

        while (value != 1L && count <= 500) {
            if (value % 2 == 0L) {
                value /= 2
            } else {
                value *= 3
                value += 1
            }
            count++
        }

        return if (count <= 500) count else -1
    }
}